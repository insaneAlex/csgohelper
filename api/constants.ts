export const inventoryImageBaseUrl =
  "https://steamcommunity-a.akamaihd.net/economy/image/";

export const getInventoryEndpoint = "http://localhost:5000/v1/csgoInventory";
export const getItemPriceEndpoint = "http://localhost:5000/v1/itemPrice";
