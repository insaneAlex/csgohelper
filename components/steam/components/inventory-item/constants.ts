export const PRIORITY_ITEMS_COUNT = 50;
