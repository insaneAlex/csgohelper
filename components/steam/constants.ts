export const inventoryImageBaseUrl =
  "https://steamcommunity-a.akamaihd.net/economy/image/";

export const inventoryBase = "https://steamcommunity.com/inventory";

export const inventoryRest = "730/2?l=english&count=1000";
