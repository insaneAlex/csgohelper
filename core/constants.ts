export const STEAMID_PARAM = 'steamid';

export const fetchInventoryUrl = '/api/csInventory';
export const postFeedbackUrl = '/api/feedback';
