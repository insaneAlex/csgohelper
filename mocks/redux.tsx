import React from 'react';

export const createStubComponent = () => (props: any) => {
  return Object.keys(props)
    .sort()
    .map((el) => <React.Fragment key={el}>{el}</React.Fragment>);
};
