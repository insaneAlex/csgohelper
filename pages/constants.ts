export const UNSTACK_DUPES = "UNSTACK DUPES";
