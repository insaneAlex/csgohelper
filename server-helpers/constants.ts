export const THIRD_OF_THE_DAY = 8 * 60 * 60 * 1000;
