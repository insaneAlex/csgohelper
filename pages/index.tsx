import {inventoryStatusSelector, itemsSelector, getItemsStart, RootState} from '@/src/redux';
import {GetInventoryPayloadType, InventoryStatuses} from '@/src/redux/features';
import {SearchInventory, InventoryLayout} from '@/src/components/steam';
import {InventoryItemType} from '@/src/services/steam-inventory';
import {ErrorAlert} from '@/src/components/ui';
import {storage} from '@/src/services';
import {useRouter} from 'next/router';
import {STEAMID_PARAM} from '@/core';
import {connect} from 'react-redux';
import {FC, useEffect} from 'react';

type Props = {
  onGetItems: (a: GetInventoryPayloadType) => void;
  inventoryItems: InventoryItemType[];
  status: InventoryStatuses;
};

export const SteamInventory: FC<Props> = ({onGetItems, inventoryItems, status}) => {
  const router = useRouter();
  const steamid = storage.localStorage.get(STEAMID_PARAM);
  const hasNoItems = inventoryItems.length === 0;

  useEffect(() => {
    hasNoItems && steamid && onGetItems({steamid});
  }, []);

  const renderError = () => {
    switch (status) {
      case InventoryStatuses.PRIVATE_INVENTORY:
        return <ErrorAlert>Inventory is private, change your privacy settings or try another account</ErrorAlert>;
      case InventoryStatuses.NO_PROFILE:
        return <ErrorAlert>There is not such profile, try another SteamID</ErrorAlert>;
    }
  };

  return (
    <>
      <SearchInventory inventoryStatus={status} />
      {renderError()}
      <InventoryLayout router={router} items={inventoryItems} />
    </>
  );
};

const mapStateToProps = (state: RootState) => ({
  status: inventoryStatusSelector(state),
  inventoryItems: itemsSelector(state)
});
const mapDispatchToProps = {onGetItems: getItemsStart};

export const SteamInventoryContainer = connect(mapStateToProps, mapDispatchToProps);
export default SteamInventoryContainer(SteamInventory);
