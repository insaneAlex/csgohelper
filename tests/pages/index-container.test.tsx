import {INVENTORY_KEY} from '@/src/redux/constants';
import {createStubComponent} from '@/mocks/redux';
import {SteamInventoryContainer} from '@/pages';
import {configureStore} from '@reduxjs/toolkit';
import {render} from '@testing-library/react';
import {inventoryReducer} from '@/src/redux';
import {Provider} from 'react-redux';
import React from 'react';

jest.mock('../../src/services', () => ({}));

describe('SteamInventoryContainer', () => {
  const mockStore = configureStore({reducer: {[INVENTORY_KEY]: inventoryReducer}});
  it('should successfully pass props', () => {
    const StubComponent = createStubComponent();
    const WrapedComponent = SteamInventoryContainer(StubComponent);
    const {container} = render(
      <Provider store={mockStore}>
        <WrapedComponent />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
