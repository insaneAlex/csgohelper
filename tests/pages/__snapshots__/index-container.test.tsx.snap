// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SteamInventoryContainer should successfully pass props 1`] = `
<div>
  inventoryItems
  onGetItems
  status
</div>
`;
